testgrid/tgrun/pkg/runner/runner.go

```diff
@@ -115,145 +115,20 @@ func execute(singleTest types.SingleRun, uploadProxyURL, tempDir string) error {
-				"testgrid.kurl.sh/osname":      singleTest.OperatingSystemName,
-				"testgrid.kurl.sh/osversion":   singleTest.OperatingSystemVersion,
-				"testgrid.kurl.sh/osimage":     singleTest.OperatingSystemImage,
-				"testgrid.kurl.sh/kurlurl":     singleTest.KurlURL,
-				"testgrid.kurl.sh/apiendpoint": singleTest.TestGridAPIEndpoint,
-			},
-		},
-		Spec: kubevirtv1.VirtualMachineInstanceSpec{
-			Domain: kubevirtv1.DomainSpec{
-				Machine: &kubevirtv1.Machine{
-					Type: "",
-				},
-				Resources: kubevirtv1.ResourceRequirements{
-					Requests: corev1.ResourceList{
-						corev1.ResourceName(corev1.ResourceMemory): resource.MustParse("16Gi"),
-						corev1.ResourceName(corev1.ResourceCPU):    resource.MustParse("4"),
-					},
-				},
-				Devices: kubevirtv1.Devices{
-					Disks: []kubevirtv1.Disk{
-						{
-							Name: "pvcdisk",
-							DiskDevice: kubevirtv1.DiskDevice{
-								Disk: &kubevirtv1.DiskTarget{
-									Bus: "virtio",
-								},
-							},
-						},
-						{
-							Name: "cloudinitdisk",
-							DiskDevice: kubevirtv1.DiskDevice{
-								CDRom: &kubevirtv1.CDRomTarget{
-									Bus: "sata",
-								},
-							},
-						},
-						getEmptyDisk("emptydisk1"),
-					},
-				},
-			},
-			TerminationGracePeriodSeconds: &zero,
-			Volumes: []kubevirtv1.Volume{
-				{
-					Name: "pvcdisk",
-					VolumeSource: kubevirtv1.VolumeSource{
-						PersistentVolumeClaim: &kubevirtv1.PersistentVolumeClaimVolumeSource{
-							PersistentVolumeClaimVolumeSource: corev1.PersistentVolumeClaimVolumeSource{
-								ClaimName: singleTest.PVCName,
-							},
-						},
-					},
-				},
-				{
-					Name: "cloudinitdisk",
-					VolumeSource: kubevirtv1.VolumeSource{
-						CloudInitNoCloud: &kubevirtv1.CloudInitNoCloudSource{
-							UserDataSecretRef: &corev1.LocalObjectReference{
-								Name: fmt.Sprintf("cloud-init-%s", singleTest.ID),
-							},
-						},
-					},
-				},
-				getEmptyDiskVolume("emptydisk1", resource.MustParse("50Gi")),
-			},
-		},
-	}
-
-	// it would be nice to add the client-go from kubevirt, but
-	// go mod is making it really hard
-	b, err := json.Marshal(vmi)
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal vmi")
-	}
-	if err := ioutil.WriteFile(filepath.Join(tempDir, "vmi.yaml"), b, 0644); err != nil {
-		return errors.Wrap(err, "failed to write file")
-	}
-
-	// mark the instance started
-	// we do this after the data volume is uploaded
-	if err := reportStarted(singleTest); err != nil {
-		return errors.Wrap(err, "failed to report test started")
-	}
-
-	cmd = exec.Command("kubectl",
-		"apply",
-		"-f",
-		filepath.Join(tempDir, "vmi.yaml"),
-	)
-
-	output, err = cmd.CombinedOutput()
-	if err != nil {
-		fmt.Printf("%s\n", output)
-		return errors.Wrap(err, "kubectl apply vmi failed")
+	createVmi(singleTest, "initialmaster", tempDir, osImagePath, uploadProxyURL)
+	for i := 0; i < 1; i++ {
+		createVmi(singleTest, "workernode", tempDir, osImagePath, uploadProxyURL)
```

Comment:
should you handle the error here?

Reply:
there are additional masters and additional workers. how can you differentiate between the two?